plugins/vendor/zoom/infra/kZoomClient.php

```diff
@@ -258,7 +261,7 @@ protected function generateContextualUrl($apiPath)
 		$url = $this -> zoomBaseURL . $apiPath . '?';
 		if ($this->refreshToken)
 		{
-			if (!$this->accessToken)
+			if (!$this->accessToken || ($this->expiresIn && $this->expiresIn <= time()))
 			{
 				$this->refreshTokens();
```

Comment:
inside refresh tokens make sure expiry is being set
also make sure there are no additional floes that set/init the access token as we will need to set there the expiry as well